source/services/environment/environmentService.ts

```diff
@@ -0,0 +1,75 @@
+import env from "react-native-config";
+
+import type {
+  EnvironmentConfig,
+  EnvironmentService,
+  RawEnvironmentConfigMap,
+} from "./environmentService.types";
+
+export default class DefaultEnvironmentService implements EnvironmentService {
+  envSource: Record<string, string> = {};
+
+  environments: Record<string, EnvironmentConfig> = {};
+
+  activeEnvironment: EnvironmentConfig | null = null;
+
+  constructor(envSource: Record<string, string> = env) {
+    this.envSource = envSource;
+  }
```

Comment:
It's not exactly clear that you need to use the `parse` function before using the setActive, function as an example.

How about setting environments on class instantiation like:

```js
constructor(environmentSource, rawConfigMap) {
  this.environmentSource = environmentSource;
  this.environments = this.parse(rawConfigMap);
}
```

Then I think the usage would be simpler:

```js
const service = new DefaultEnvironmentService(env, configMap);
service.setActive(environmentName);
```

And we also avoid throwing an error message that looks like this:
```js
Environment '<ENVIRONMENT>' does not exist in list: ""
```

which could be the case if `parse` is not run before `setActive`